pkg/promclient/promclient.go

```diff
@@ -362,3 +406,78 @@ func convertScalarJSONToVector(scalarJSONResult json.RawMessage) (model.Vector,
 		Value:     resultValue,
 		Timestamp: resultTime}}, nil
 }
+
+// MetricValues returns current value of  instant query and returns results in model.Vector type.
+func MetricValues(ctx context.Context, logger log.Logger, base *url.URL, perMetricFn func(metric promlabels.Labels, val float64) error) error {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+
+	u := *base
+	u.Path = path.Join(u.Path, "/metrics")
+
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return errors.Wrap(err, "create GET request")
+	}
+
+	req.Header.Add("Accept", `application/openmetrics-text; version=0.0.1,text/plain;version=0.0.4;q=0.5,*/*;q=0.1`)
+	req.Header.Add("Accept-Encoding", "gzip")
+
+	req = req.WithContext(ctx)
+
+	client := &http.Client{
+		Transport: tracing.HTTPTripperware(logger, http.DefaultTransport),
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.Wrapf(err, "perform GET request against %s", u.String())
+	}
+	defer runutil.CloseWithLogOnErr(logger, resp.Body, "metrics body")
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Errorf("server returned HTTP status %s", resp.Status)
+	}
+
+	b := &bytes.Buffer{}
+	if resp.Header.Get("Content-Encoding") != "gzip" {
+		_, err = io.Copy(b, resp.Body)
+		if err != nil {
+			return err
+		}
+	} else {
+		buf := bufio.NewReader(resp.Body)
+		gzipr, err := gzip.NewReader(buf)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(b, gzipr)
+		_ = gzipr.Close()
```

Comment:
Also don't we need to close `resp.Body` here?

Reply:
same here, copied, but maybe badly, relooking

Reply:
433 line is already defering close